### ISSUE TITLE
BaseLocationServices is not assignable to LocationServices

### ISSUE THREAD

**Issue by robertomatute** [ui-router/angular-hybrid#197]
Hi,

I was using `@ui-router/angular-hybrid` (v 5.0.0) and everything run fine.  I could change from states in my angularjs app, to my angular.io app without any problem.

I just upgraded from  to 6.0.2, and I getting this error in the compilation:
```
ERROR in node_modules/@uirouter/core/lib/vanilla/baseLocationService.d.ts(43,5): error TS2416: Property 'onChange' in type 'BaseLocationServices' is not assignable to the same property in base type 'LocationServices'.
  Type '(cb: EventListener) => () => Function[]' is not assignable to type '(callback: Function) => Function'.
    Types of parameters 'cb' and 'callback' are incompatible.
      Type 'Function' is not assignable to type 'EventListener'.
        Type 'Function' provides no match for the signature '(evt: Event): void'.
```

The app load in the browser and it works fine, without any error in the console.

Anyone with this problem?

Thanks!!

**Comment by robertomatute** [comment]
Hi,

Browsing the core issues, I find the same issue:
https://github.com/ui-router/core/issues/229

Regards

**Comment by christopherthielen** [comment]
fixed in latest core release `5.0.23`
